gostream/media.go

```diff
@@ -284,7 +285,9 @@ func (pc *producerConsumer[T, U]) start() {
 					doReadSpan.End()
 				}()
 
-				var lastRelease func()
+				if lastRelease != nil {
```

Comment:
What was wrong about where `lastRelease` was before?

Reply:
Good point. If I move it back to where it was. We also don't need the lastRelease() after the loop